code/modules/client/preferences2/preferences2.dm

```diff
@@ -0,0 +1,230 @@
+#define PREP_WRITEPREF_JSONENC(value, tag) PREP_WRITEPREF_RAW(json_encode(value), tag)
+
+
+// Writes all prefs to the DB
+/datum/preferences/proc/save_preferences()
+	if(!SSdbcore.IsConnected())
+		return
+
+	if(IS_GUEST_KEY(parent.ckey))
+		return
+
+	var/list/datum/DBQuery/write_queries = list() // do not rename this you muppet
+
+	//general preferences
+	PREP_WRITEPREF_RAW(default_slot, PREFERENCE_TAG_DEFAULT_SLOT)
+	PREP_WRITEPREF_RAW(chat_toggles, PREFERENCE_TAG_CHAT_TOGGLES)
+	PREP_WRITEPREF_RAW(toggles, PREFERENCE_TAG_TOGGLES)
+	PREP_WRITEPREF_RAW(toggles2, PREFERENCE_TAG_TOGGLES2)
+	PREP_WRITEPREF_RAW(clientfps, PREFERENCE_TAG_CLIENTFPS)
+	PREP_WRITEPREF_RAW(parallax, PREFERENCE_TAG_PARALLAX)
+	PREP_WRITEPREF_RAW(pixel_size, PREFERENCE_TAG_PIXELSIZE)
+	PREP_WRITEPREF_RAW(tip_delay, PREFERENCE_TAG_TIP_DELAY)
+	PREP_WRITEPREF_RAW(pda_style, PREFERENCE_TAG_PDA_STYLE)
+	PREP_WRITEPREF_RAW(pda_color, PREFERENCE_TAG_PDA_COLOUR)
+
+	PREP_WRITEPREF_RAW(asaycolor, PREFERENCE_TAG_ASAY_COLOUR)
+	PREP_WRITEPREF_RAW(ooccolor, PREFERENCE_TAG_OOC_COLOUR)
+	PREP_WRITEPREF_RAW(lastchangelog, PREFERENCE_TAG_LAST_CL)
+	PREP_WRITEPREF_RAW(UI_style, PREFERENCE_TAG_UI_STYLE)
+	PREP_WRITEPREF_RAW(outline_color, PREFERENCE_TAG_OUTLINE_COLOUR)
+	PREP_WRITEPREF_RAW(see_balloon_alerts, PREFERENCE_TAG_BALLOON_ALERTS)
+	PREP_WRITEPREF_RAW(scaling_method, PREFERENCE_TAG_SCALING_METHOD)
+	PREP_WRITEPREF_RAW(ghost_form, PREFERENCE_TAG_GHOST_FORM)
+	PREP_WRITEPREF_RAW(ghost_orbit, PREFERENCE_TAG_GHOST_ORBIT)
+	PREP_WRITEPREF_RAW(ghost_accs, PREFERENCE_TAG_GHOST_ACCS)
+	PREP_WRITEPREF_RAW(ghost_others, PREFERENCE_TAG_GHOST_OTHERS)
+
+	PREP_WRITEPREF_JSONENC(ignoring, PREFERENCE_TAG_IGNORING)
+	PREP_WRITEPREF_JSONENC(key_bindings, PREFERENCE_TAG_KEYBINDS)
+	PREP_WRITEPREF_JSONENC(purchased_gear, PREFERENCE_TAG_PURCHASED_GEAR)
+	PREP_WRITEPREF_JSONENC(be_special, PREFERENCE_TAG_BE_SPECIAL)
+
+	// QuerySelect can execute many queries at once. That name is dumb but w/e
+	SSdbcore.QuerySelect(write_queries, TRUE, TRUE)
+
+#undef PREP_WRITEPREF_RAW
+#undef PREP_WRITEPREF_JSONENC
+
+
+// Get ready for a disgusting SQL query
+/datum/preferences/proc/load_characters()
+	// Do NOT remove stuff from the start of this query. Only append to the end.
+	// If you delete an entry, god help you as you have to update all the indexes
+	var/datum/DBQuery/read_chars = SSdbcore.NewQuery({"
+		SELECT
+			slot,
+			species,
+			real_name,
+			name_is_always_random,
+			body_is_always_random,
+			gender,
+			age,
+			hair_color,
+			gradient_color,
+			facial_hair_color,
+			skin_tone,
+			hair_style_name,
+			gradient_style,
+			facial_style_name,
+			underwear,
+			underwear_color,
+			undershirt,
+			socks,
+			backbag,
+			jumpsuit_style,
+			uplink_loc,
+			features,
+			custom_names,
+			helmet_style,
+			preferred_ai_core_display,
+			preferred_security_department,
+			joblessrole,
+			job_preferences,
+			all_quirks,
+			equipped_gear
```

Comment:
this is missing eye_color between facial_hair_color and skin_tone

Reply:
Tested... Adding in the missing column brings the total number of items to the expected 31